File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/builder/Director.java
package jv.com.tencent.cain.design_pattern.builder;

/**
 * @author : jiangyu
 * @date : 2020/2/10
 * @desc : xxx
 */
public class Director {

    public void construct(Builder builder) {
        builder.buildPartA();
        builder.buildPartB();
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/builder/Director.kt
package kt.com.tencent.cain.desigin_pattern.builder

/**
 * @author : jiangyu
 * @date   : 2020/2/10
 * @desc   : xxx
 */
class Director {

    fun construct(builder: Builder) {
        builder.buildPartB()
        builder.buildPartA()
    }
}